lib/puppet/confine/false.rb

```diff
@@ -1,6 +1,8 @@
-require 'puppet/confine'
+require 'puppet/confine/boolean'
 
 class Puppet::Confine::False < Puppet::Confine
+  include Puppet::Confine::Boolean
```

Comment:
The reason I'm using mixin here instead of inheritance is because of https://github.com/ekinanp/puppet/blob/PUP-9336/lib/puppet/confine.rb#L14-L38. Specifically, if I inherit from `Puppet::Confine::Boolean`, then that class would have its own `@tests` variable which is different from `Puppet::Confine` so that the logic in the [confine selection](https://github.com/ekinanp/puppet/blob/PUP-9336/lib/puppet/confine_collection.rb#L14-L17) would break.

Reply:
Opened [PUP-9367](https://tickets.puppetlabs.com/browse/PUP-9367) about fixing how `Puppet::Confine` does the registration, and duplicate checking to fix this issue.

Reply:
I hate that that is even possible in ruby... In that case, what about having the mixin-ee code `Puppet::Confine::Boolean` call a method `passing_value`, and have the mixin-er implement that method?